### PR TITLE
Navigation: Add scroll indicators

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

# before
https://user-images.githubusercontent.com/20999846/164715887-3f5258fe-2a36-448a-aadf-cc0758b37a6d.mp4

# after 
https://user-images.githubusercontent.com/20999846/164715924-46adf691-b40e-44a0-b5ee-925072472c81.mp4

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  }),
+  scrollBottomMarker: css({
+    bottom: 0,
+  }),
+  scrollContent: css({
+    position: 'relative',
+  }),
+  scrollContainer: css`
+    .scrollbar-view {
+      position: static !important;
```
Comment (JoaoSilvaGrafana) [inline public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:84]: Could you add a comment here for why this `!important` is needed? Always wary of these 😅

File: public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    bottom: 0,
+    height: theme.spacing(6),
+    left: 0,
+    opacity: 0,
+    pointerEvents: 'none',
+    position: 'absolute',
+    right: 0,
+    transition: theme.transitions.create('opacity'),
+    zIndex: theme.zIndex.sidemenu,
+  }),
```
Comment (JoaoSilvaGrafana) [inline public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:110]: Could you do something similar to what you did for `'scrollTopMarker, scrollBottomMarker'` and join the common styling and just keep what is different separate? I find it easier to understand then what are the differences between the two

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Great job, looks so smooth! Comments I left are more nitpicky.

I did find one strange scenario, if you tab into a menu item that isn't visible, it will scroll to it, but not expand the menu 😞 


https://user-images.githubusercontent.com/100691367/164730890-c3f87c01-aa66-43bb-bcf6-695ea1a3ddef.mov

File: public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx (hunk 2 above)
Comment (ashharrison90) [reply public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:110]: @JoaoSilvaGrafana i tried that but it doesn't work - i actually don't know why tbh as it was exactly the same syntax as the other... 😢

**Comment by ashharrison90** [comment]
> I did find one strange scenario, if you tab into a menu item that isn't visible, it will scroll to it, but not expand the menu

yeah i'm not sure how we're gonna fix that nicely. 😕  guess it's not specific to this PR though since it occurs on main as well

File: public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:88]: yeah i don't like it either. if we can find a better way i'd love to change it

Comment (JoaoSilvaGrafana) [reply public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:88]: Yeah that's fair, thanks for adding the comment!

**Comment by JoaoSilvaGrafana** [comment]
> > I did find one strange scenario, if you tab into a menu item that isn't visible, it will scroll to it, but not expand the menu
> 
> yeah i'm not sure how we're gonna fix that nicely. 😕 guess it's not specific to this PR though since it occurs on main as well

Oh yeah didn't notice that, I can do a bit of investigation into what might be happening with the focus hook

File: public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx (hunk 2 above)
Comment (JoaoSilvaGrafana) [reply public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx:114]: Weird! Nevermind, it was just a small thing anyway

**Review by JoaoSilvaGrafana — APPROVED** [review]
Looks good to me, I'll investigate that tabbing thing then!

**Pull request merged**